### ISSUE TITLE
Inconsistency in `kmeans.labels_` attribute

### ISSUE THREAD

**Issue by OSuwaidi** [gittar/bkmeans#4]
When fitting `BKMeans` on some toy data:

```
data = {'Feature1': np.random.randint(0, 10, (50,)),
        'Feature2': np.random.randint(0, 10, (50,))}
df = pd.DataFrame(data)

kmeans = BKMeans(n_clusters=4, random_state=0)
kmeans.fit(df)
```

One would expect the result of `kmeans.predict(df)` to match the result of `kmeans.labels_`, but that's not the case!

What's more, is that every time you rerun the `BKMeans()` fit on the same data with the same initialization, the `kmeans.labels_` result changes every time, while `kmeans.predict(df)` remains the same.

**Comment by gittar** [comment]
thx. Just confirmed this behavior. Will analyze and fix.

**Comment by gittar** [comment]
fix committed in Line 91:
self.labels_ = self.predict(X)

**Comment by OSuwaidi** [comment]
Hey @gittar,

That should be it, works great thanks a lot!

**Comment by OSuwaidi** [comment]
@gittar Would you like me to create a pull request for this?

**Comment by gittar** [comment]
That would be excellent!

**Comment by gittar** [comment]
could you provide a sample prog + output illustrating the behavior?

**Comment by gittar** [comment]
> When fitting `BKMeans` on some toy data:
> 
> ```
> data = {'Feature1': np.random.randint(0, 10, (50,)),
>         'Feature2': np.random.randint(0, 10, (50,))}
> df = pd.DataFrame(data)
> 
> kmeans = BKMeans(n_clusters=4, random_state=0)
> kmeans.fit(df)
> ```
> 
> One would expect the result of `kmeans.predict(df)` to match the result of `kmeans.labels_`, but that's not the case!
> 
> What's more, is that every time you rerun the `BKMeans()` fit on the same data with the same initialization, the `kmeans.labels_` result changes every time, while `kmeans.predict(df)` remains the same.

both points fixed in new version 1.3
